front/app/containers/Admin/projects/project/general/components/GeographicAreaInputs.tsx

```diff
@@ -1,78 +1,148 @@
+import { isNilOrError, isString } from 'utils/helperUtils';
 import { IconTooltip, Radio } from '@citizenlab/cl2-component-library';
 import Link from 'utils/cl-router/Link';
 import { SubSectionTitle } from 'components/admin/Section';
 import { StyledSectionField, StyledMultipleSelect } from './styling';
-
-// i18n
 import { FormattedMessage } from 'utils/cl-intl';
 import messages from '../messages';
-
-// typings
-import { IOption } from 'typings';
+import { IOption, isOption } from 'typings';
+import useAreas from 'hooks/useAreas';
+import useProject from 'hooks/useProject';
+import useLocalize from 'hooks/useLocalize';
+import { IAreaData } from 'services/areas';
+import { useParams } from 'react-router-dom';
+import { TOnProjectAttributesDiffChangeFunction } from '..';
 
 interface Props {
-  areaType: 'all' | 'selection';
-  areasOptions: IOption[];
-  areasValues: IOption[];
-  handleAreaTypeChange: (value: 'all' | 'selection') => void;
-  handleAreaSelectionChange: (values: IOption[]) => void;
+  areaIds: string[] | undefined;
+  onProjectAttributesDiffChange: TOnProjectAttributesDiffChangeFunction;
 }
 
-export default ({
-  areaType,
-  areasOptions,
-  areasValues,
-  handleAreaTypeChange,
-  handleAreaSelectionChange,
-}: Props) => (
-  <StyledSectionField>
-    <SubSectionTitle>
-      <FormattedMessage {...messages.areasLabel} />
-      <IconTooltip
-        content={
-          <FormattedMessage
-            {...messages.areasLabelTooltip}
-            values={{
-              areasLabelTooltipLink: (
-                <Link to="/admin/settings/areas">
-                  <FormattedMessage {...messages.areasLabelTooltipLinkText} />
-                </Link>
-              ),
-            }}
+type TProjectAreaType = 'all' | 'selection';
+
+const GeographicAreaInputs = ({
+  areaIds,
+  onProjectAttributesDiffChange,
+}: Props) => {
+  const { projectId } = useParams();
+  const areas = useAreas();
+  const project = useProject({ projectId });
+  const localize = useLocalize();
+  const [areaType, setAreaType] = useState<TProjectAreaType>('all');
+
+  const handleAreaSelectionChange = (values: IOption[]) => {
+    const selectedAreaIds = values.map((value) => value.value).filter(isString);
+    onProjectAttributesDiffChange({ area_ids: selectedAreaIds });
+  };
+
+  const handleAreaTypeOnChange = (areaType: TProjectAreaType) => {
+    setAreaType(areaType);
+    onProjectAttributesDiffChange({
+      area_ids: areaType === 'all' ? [] : areaIds,
+    });
+  };
+
+  const mapProjectAreaIdsToAreaOptions = (
+    projectAreaIds: string[],
+    areas: IAreaData[]
+  ) => {
+    projectAreaIds
+      .map((projectAreaId) => {
+        const projectArea = getProjectArea(areas, projectAreaId);
+
+        return projectArea
+          ? {
+              value: projectAreaId,
+              label: localize(projectArea.attributes.title_multiloc),
+            }
+          : null;
+      })
+      .filter(isOption);
```

Comment:
This will make sure the types say we return IOption[], not (IOption | null[]

Normally, there will be no undefined projectArea and no areaOptions that are null, but TS doesn't know and I don't want to enforce these types because it's still possible to have such values.